packaging/overview.md

```diff
@@ -0,0 +1,50 @@
+# Code Packaging
+
+Learning goals:
+
+- Understanding the necessity to somehow package code written in a crude way
+- Being able to determine the type of packaging required for the code
+- Modifying code and packaging it according to a standardization
+- Uploading code to a remote package managing platform and testing the installation
+
+## Survey
+
+| Duration | Format | Material |
+| --- | --- | --- |
+| 15 minutes | poll | [packaging_survey.md](https://github.com/Simulation-Software-Engineering/Lecture-Material) |
+
+- Conduct an online poll with four questions to get an impression of how much the students already know about packaging
+- Discuss results and quantify the minimum knowledge level of the class
+- **Most important question**: How do students perceive the term *packaging*?
+
+## Introduction to packaging
+
+| Duration | Format | Material |
+| --- | --- | --- |
+| 20 minutes | slides | [intro_slides.md](https://github.com/Simulation-Software-Engineering/Lecture-Material) |
+
+## How to prepare a Python code for packaging
+
+| Duration | Format | Material |
+| --- | --- | --- |
+| 40 minutes | slides | [packaging_python.md](https://github.com/Simulation-Software-Engineering/Lecture-Material) |
+
+## Exercise
+
+### Part 1: Prepare a raw code for packaging
+
+| Duration | Format | Material |
+| --- | --- | --- |
+| 50 minutes | in-class exercise | [exercise_packaging_part1.md](https://github.com/Simulation-Software-Engineering/Lecture-Material) |
+
+### Part 2: Upload code to TestPyPI and install it using pip
+
+| Duration | Format | Material |
+| --- | --- | --- |
+| 40 minutes | in-class exercise | [exercise_packaging_part2.md](https://github.com/Simulation-Software-Engineering/Lecture-Material) |
+
+## Further reading
+
+- [The Hitchhiker's Guide to Python](https://docs.python-guide.org/shipping/packaging/)
+- [readthedocs guide](https://python-packaging.readthedocs.io/en/latest/)
+- [Python Packaging User Guide](https://packaging.python.org/)
```

Comment:
- [pypi help](https://pypi.org/help/)?
- Shouldn't the file end with a new line?